.github/workflows/mutants.yml

```
@@ -32,40 +74,80 @@ jobs:
 
       - uses: ./.github/actions/rust
         with:
+          version: nightly
           tools: cargo-mutants
           token: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Find incremental mutants
-        id: mutants
+      - name: Find mutants
         env:
-          BASE_REF: ${{ github.base_ref }}
-        if: ${{ github.event_name == 'pull_request' }}
+          SHARD: ${{ matrix.shard }}/${{ strategy.job-total }}
+          TIMEOUT: ${{ needs.baseline.outputs.timeout }}
         run: |
-          git diff "origin/$BASE_REF".. > pr.diff
           set -o pipefail
-          cargo mutants --no-shuffle -j 2 -vV --in-diff pr.diff | tee results.txt || true
-          echo 'title=Incremental Mutants' >> "$GITHUB_OUTPUT"
+          # shellcheck disable=SC2086
+          # Exit codes: 0=ok, 2=missed, 3=timeout, 4=unviable (all expected).
+          cargo mutants --shard "$SHARD" --sharding round-robin --baseline=skip --timeout "$TIMEOUT" $MUTANTS_ARGS 2>&1 | tee results.txt \
+            || { ec=$?; [ $ec -le 4 ] && exit 0; exit $ec; }
```

You don't need pipefail if you shuffle...


```suggestion
          (cargo mutants --shard "$SHARD" --sharding round-robin --baseline=skip --timeout "$TIMEOUT" $MUTANTS_ARGS 2>&1 || { ec=$?; [ $ec -le 4 ] && exit 0; exit $ec }) | tee results.txt
```